app/processing/process-payment-request.js

```diff
@@ -18,6 +20,17 @@ const processPaymentRequest = async (scheduledPaymentRequest) => {
 
   const paymentRequests = await transformPaymentRequest(paymentRequest)
   const { deltaPaymentRequest, completedPaymentRequests } = paymentRequests
+
+  // if FRN is closed (SFI only), remove AR
+  if (paymentRequest.schemeId === SFI && config.handleSFIClosures) {
+    const closedFRN = await getClosedFRN(paymentRequest.frn)
+    if (closedFRN !== null) {
+      console.log(`FRN ${paymentRequest.frn} has been closed, skipping request editor and holds`)
+      await mapAndComplete(scheduleId, completedPaymentRequests)

```

Comment:
Skipping straight to this stage doesn't result in the intended behaviour.

After `transformPaymentRequest` we will get an array of payment requests under the property `completedPaymentRequests`.  Those payment requests will all have a ledger of either `AP` or `AR`. 

If the payment request is SFI 22 and FRN is on the closures list (probably also matches a closure date and agreement - see above) then we need to remove all the AR payments requests from the array.

We need to skip the Request Editor.  Don't necessarily need to skip the auto hold functionality.  In fact we probably want to allow that to happen should it be needed.

Reply:
behaviour amended to simply filter out the completedPaymentRequests for anything stating AR. logic added specifically to skip RE if the closure function returns true, holds is not skipped.